src/olympia/blocklist/mlbf.py

```diff
@@ -136,50 +169,46 @@ def __init__(self, *args, **kwargs):
 
     @cached_property
     def _all_blocks(self):
-        return (
-            BlockVersion.objects.filter(version__file__is_signed=True)
-            .distinct()
-            .order_by('id')
-            .values_list(
-                'block__guid',
-                'version__version',
-                'version_id',
-                'block_type',
-                named=True,
-            )
-        )
+        _blocked_version_ids = []
+        _blocked = {
+            BlockType.BLOCKED: [],
+            BlockType.SOFT_BLOCKED: [],
+        }
 
-    def _format_blocks(self, block_type: BlockType) -> List[str]:
-        return MLBF.hash_filter_inputs(
-            [
-                (version.block__guid, version.version__version)
-                for version in self._all_blocks
-                if version.block_type == block_type
-            ]
-        )
+        # We define get_all_blocked_items as a separate function to allow
+        # mocking the database query in tests to simulate large data sets.
+        for guid, version, id, block_type in get_all_blocked_items():
+            _blocked_version_ids.append(id)
+            _blocked[block_type].append((guid, version))
```

Comment:
```suggestion
        for guid, version_string, version_id, block_type in get_all_blocked_items():
            _blocked_version_ids.append(version_id)
            _blocked[block_type].append((guid, version_string))
```

Reply:
Though I do wonder if it's more efficient to generate `_blocked_version_ids` and `_blocked` with list/set generator syntax, rather than loop through with a for loop and append one at a time.

Reply:
I was thinking more than, in python anyway, it's generally more efficient to build a list with a generator than to build it by iterating and repeatedly appending items (because it needs resizing and reallocating more).  I don't have stats to back-up the difference between generating _two_ lists vs. iterating over the list once though.

Reply:
Yeah IDK about the perf difference there, but I imagine it would not be significant since we are comparing a slightly slower single iteration versus a slightly faster double iteration. We could give it a try and see if there is a big diff.